collect

```diff
@@ -5,6 +5,7 @@ Bundler.setup(:default)
 
 $LOAD_PATH << "./lib"
 
+require "exception_handler"
 require "govuk_sli_collector"
 
-GovukSliCollector.call
+ExceptionHandler.report { GovukSliCollector.call }
```

Comment:
Is the report method needed at all? Unhandled errors should be logged in Sentry anyway

Reply:
I hoped that the Sentry gem might work that way but it didn't happen in my integration testing. Maybe I've missed a configuration option, but I haven't seen one in the docs so far.

Reply:
I'll try that now. I thought I had already, but I can only find deploys featuring all the other combinations of `Sentry.init` and raising an exception on purpose :grimacing:

Reply:
Here's how I've just tested it...

I deployed this branch https://github.com/alphagov/govuk-sli-collector/tree/install-sentry-test-handling-unhandled (the significant changes are https://github.com/alphagov/govuk-sli-collector/blob/install-sentry-test-handling-unhandled/collect#L8-L14 and https://github.com/alphagov/govuk-sli-collector/blob/install-sentry-test-handling-unhandled/lib/govuk_sli_collector.rb#L8).

I confirmed that the error was logged in the container https://argo.eks.integration.govuk.digital/applications/govuk-jobs?orphaned=false&resource=&node=%2FPod%2Fapps%2Fgovuk-sli-collector-28366080-l9v5d&tab=logs (and appears in Logit https://kibana.logit.io/s/42f4d2d5-e9ce-451f-8ffc-cdb25bd624f8/app/discover#/doc/filebeat-*/filebeat-2023.12.07?id=Su8CRYwB5HoHRDiIDzvo).

But I can only see the existing errors in Sentry (https://govuk.sentry.io/issues/?groupStatsPeriod=auto&project=4506338071150592&query=&referrer=issue-list&statsPeriod=7d) with no increase in their event counts.

Reply:
It gets that from the container's environment (I don't know if I did a good job of it, but I tried to explain this in the commit that calls `Sentry.init`: https://github.com/alphagov/govuk-sli-collector/commit/074c0919b4b711cf3ddfba0674aad7f9386c3157 in case it was a bit mysterious)

Reply:
Ah, I see, it reads them from the env vars if available.

It does say that unhandled errors should be logged, I can't see anything wrong with the install-sentry-test-handling-unhandled branch though https://docs.sentry.io/product/sentry-basics/integrate-backend/capturing-errors/#unhandled-errors